libcudacxx/include/cuda/std/detail/libcxx/include/__cccl_config

```diff
@@ -47,46 +51,44 @@
 #endif
 
 // Enforce that cccl headers are treated as system headers
-#if defined(_CCCL_COMPILER_GCC)
-#  define _CCCL_FORCE_SYSTEM_HEADER _Pragma("GCC system_header")
+#if defined(_CCCL_COMPILER_GCC) || defined(_CCCL_COMPILER_NVHPC) || defined(_CCCL_COMPILER_ICC) \
```

Comment:
This is the real change. We filter compilers and use the appropriate #pragma depending on the define